flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -157,14 +216,25 @@ public boolean reachedEnd() throws IOException {
 	 * @throws java.io.IOException
 	 */
 	@Override
-	public OUT nextRecord(OUT tuple) throws IOException {
+	public Row nextRecord(Row row) throws IOException {
 		try {
-			resultSet.next();
-			if (columnTypes == null) {
-				extractTypes(tuple);
+			hasNext = resultSet.next();
+			if (!hasNext) {
+				return null;
+			}
+			try {
+				//This throws a NPE when the TypeInfo is not passed to the InputFormat,
+				//i.e. KryoSerializer used to generate the passed row
+				row.productArity();
```

Comment:
because it can happen that the row !=null but inside it is uninitialized, for example when the  (see for example **JDBCInputFormatTest.testUninitializedRow**.

That happens when you do the following (for example):
``` java
DataSet<Row> source = environment.createInput(inputBuilder.finish());
```

 The only method I found to detect such a situation is to catch the nullPointerException generated when trying to access to one of its methods, e.g. 
```java
row.productArity()
```